src/blackboxTest/bash/BlackBoxTest.sh

```diff
@@ -27,39 +28,18 @@ dispayResults() {
   fi
 }
 
-testResults() {
-  printf "\n\n${WHITE}Black Box Test > $1"
-  if cmp -s ${TESTPATH}/$2 ${TESTPATH}/$3 ; then
-    printf "${GREEN} PASSED ${WHITE}\n"
-  else
-    EXIT_CODE=1
-    printf "${RED} FAILED ${WHITE}\n"
-    printf "\n\n--------------\n"
-    diff ${TESTPATH}/$2 ${TESTPATH}/$3
-    printf "\n\n--------------\n"
-  fi
-  rm ${TESTPATH}/$2
-  rm ${TESTPATH}/$3
+updateExitCode() {
+if [ $? -eq 1 ]; then
```

Comment:
Not all programs return a `1` on failure. Typically the convention is 0 on success and non-0 on failure. Maybe a shell equivalent of `result ||= $?` would be better here–would get rid of the if check.